test/CommonLibTests/Helpers.cpp

```diff
@@ -37,12 +14,28 @@ Helpers::ReadFileContent(std::wstring file)
 
     int nChars = MultiByteToWideChar(CP_ACP,  0, buffer.str().c_str(), -1, NULL, 0);
 
-    LPWSTR pwzName = new WCHAR[nChars];
-    MultiByteToWideChar(CP_UTF8, 0, buffer.str().c_str(), -1, pwzName, nChars);
-
-    std::wstring retVal(pwzName);
+    std::wstring retVal(nChars, '\0');
 
-    delete pwzName;
+    MultiByteToWideChar(CP_UTF8, 0, buffer.str().c_str(), -1, retVal.data(), nChars);
 
     return retVal;
 }
+
+TempDirectory::TempDirectory()
+{
+    UUID uuid;
+    UuidCreate(&uuid);
+    RPC_CSTR szUuid = NULL;
+    if (UuidToStringA(&uuid, &szUuid) == RPC_S_OK)
+    {
+        m_path = std::experimental::filesystem::temp_directory_path() / szUuid;
```

Comment:
make it != RPC_S_OK and flip the contents of the if with the throw.

Reply:
I'll do it in another PR, this was just too green.